source/plugins/go/src/extension/extension.go

```diff
@@ -23,66 +24,92 @@ var containerType string
-		logger.Printf(message)
-	}
-	return e.datatypeStreamIdMap[datatype]
-}
-
-func getDataTypeToStreamIdMapping() (map[string]string, error) {
+func getExtensionConfigs() ([]ExtensionConfig, error) {
 	guid := uuid.New()
-	datatypeOutputStreamMap := make(map[string]string)
-
 	taggedData := map[string]interface{}{"Request": "AgentTaggedData", "RequestId": guid.String(), "Tag": "ContainerInsights", "Version": "1"}
 	jsonBytes, err := json.Marshal(taggedData)
-	// TODO: this error is unhandled
 
 	var data []byte
 	enc := codec.NewEncoderBytes(&data, new(codec.MsgpackHandle))
 	if err := enc.Encode(string(jsonBytes)); err != nil {
-		return datatypeOutputStreamMap, err
+		return nil, err
 	}
 
 	fs := &FluentSocket{}
 	fs.sockAddress = "/var/run/mdsd-ci/default_fluent.socket"
 	if containerType != "" && strings.Compare(strings.ToLower(containerType), "prometheussidecar") == 0 {
 		fs.sockAddress = fmt.Sprintf("/var/run/mdsd-%s/default_fluent.socket", containerType)
 	}
+	
 	responseBytes, err := FluentSocketWriter.writeAndRead(fs, data)
 	defer FluentSocketWriter.disconnect(fs)
 	if err != nil {
-		return datatypeOutputStreamMap, err
+		return nil, err
 	}
-	response := string(responseBytes) // TODO: why is this converted to a string then back into a []byte?
-
-	var responseObjet AgentTaggedDataResponse
-	err = json.Unmarshal([]byte(response), &responseObjet)
+	var responseObject AgentTaggedDataResponse
+	err = json.Unmarshal(responseBytes, &responseObject)
 	if err != nil {
 		logger.Printf("Error::mdsd::Failed to unmarshal config data. Error message: %s", string(err.Error()))
-		return datatypeOutputStreamMap, err
+		return nil, err
 	}
 
 	var extensionData TaggedData
-	json.Unmarshal([]byte(responseObjet.TaggedData), &extensionData)
+	json.Unmarshal([]byte(responseObject.TaggedData), &extensionData)
 
-	extensionConfigs := extensionData.ExtensionConfigs
+	return extensionData.ExtensionConfigs, nil
+}
+
+func getExtensionSettings() (map[string]map[string]interface{}, error) {
+	extensionSettings := make(map[string]map[string]interface{})
+
+	extensionConfigs, err := getExtensionConfigs()
+	if err != nil {
+		return extensionSettings, err
+	}
+	for _, extensionConfig := range extensionConfigs {
+		extensionSettingsItr := extensionConfig.ExtensionSettings
+		if extensionSettingsItr != nil && len(extensionSettingsItr) > 0 {
+			extensionSettings = extensionSettingsItr
+		}
+	}
+
+	return extensionSettings, nil
+}
+
+func getDataCollectionSettings() (map[string]string, error) {
+	dataCollectionSettings := make(map[string]string)
+
+	extensionSettings, err := getExtensionSettings()
+	if err != nil {
+		return dataCollectionSettings, err
+	}
+	dataCollectionSettingsItr := extensionSettings["dataCollectionSettings"]
+	if dataCollectionSettingsItr != nil && len(dataCollectionSettingsItr) > 0 {
+		for k, v := range dataCollectionSettingsItr {
+			dataCollectionSettings[k] = fmt.Sprintf("%v", v)

```

Comment:
we should convert the keys to lowercase to handle scenario where customer has different casing for example enablecontainerlogv2 or EnableContainerLogV2  instead of enableContainerLogV2.

Reply:
converted both key and value and changed the corresponding place